src/contracts/libraries/BytesArrayBitmaps.sol

```diff
@@ -0,0 +1,197 @@
+        for (uint256 i = 0; i < bytesArray.length; ++i) {
+            // construct a single-bit mask from the numerical value of the next byte out of the array
+            bitMask = uint256(1 << uint8(bytesArray[i]));
+            // check that the entry is not a repeat
+            require(bitmap & bitMask == 0, "BytesArrayBitmaps.bytesArrayToBitmap: repeat entry in bytesArray");
+            // add the entry to the bitmap
+            bitmap = (bitmap | bitMask);
+        }
+        return bitmap;
+    }
+
+    /**
+     * @notice Converts an ordered array of bytes into a bitmap.
+     * @param orderedBytesArray The array of bytes to convert/compress into a bitmap. Must be in strictly ascending order.
+     * @return The resulting bitmap.
+     * @dev Each byte in the input is processed as indicating a single bit to flip in the bitmap.
+     * @dev This function will eventually revert in the event that the `orderedBytesArray` is not properly ordered (in ascending order).
+     */
+    function orderedBytesArrayToBitmap(bytes calldata orderedBytesArray) internal pure returns (uint256) {
+        // sanity-check on input. a too-long input would fail later on due to having duplicate entry(s)
+        require(orderedBytesArray.length <= 256, "BytesArrayBitmaps.orderedBytesArrayToBitmap: orderedBytesArray is too long");
+        // return empty bitmap early if length of array is 0
+        if (orderedBytesArray.length == 0) {
+            return uint256(0);
+        }
+
+        // initialize the empty bitmap, to be built inside the loop
+        uint256 bitmap;
+        // initialize an empty byte object, to be re-used inside the loop
+        bytes1 singleByte;
+        // initialize an empty uint256 to be used as a bitmask inside the loop
+        uint256 bitMask;
+
+        // perform the 0-th loop iteration with the ordering check *omitted* (otherwise it will break with an out-of-bounds error)
+        // pull the 0th byte out of the array
+        singleByte = orderedBytesArray[0];
+        // construct a single-bit mask from the numerical value of the byte
+        bitMask = uint256(1 << uint8(singleByte));
+        // add the entry to the bitmap
+        bitmap = (bitmap | bitMask);
+
+        // loop through each byte in the array to construct the bitmap
+        for (uint256 i = 1; i < orderedBytesArray.length; ++i) {
+            // check that the entry is *strictly greater than* the previous entry. enforces both ordering and non-duplication in the array
+            require(uint256(uint8(orderedBytesArray[i])) > uint256(uint8(singleByte)),
+                "BytesArrayBitmaps.orderedBytesArrayToBitmap: orderedBytesArray is not ordered");
+            // pull the next byte out of the array
+            singleByte = orderedBytesArray[i];
+            // construct a single-bit mask from the numerical value of the byte
+            bitMask = uint256(1 << uint8(singleByte));
```

Comment:
could we do the same check you wrote below here? just 
```solidity
bitMask = uint256(1 << uint8(orderedBytesArray[i]));
require(bitmask > bitmap)
```

Reply:
yes, I'll change that, it's definitely more efficient

Reply:
done in https://github.com/Layr-Labs/eigenlayer-contracts/pull/49/commits/857740bc449ad16c4f5e1cbca0fdc5ddfbdd9168